File: Kanban-Board-Frontend/src/app/interface/DeletedTask.ts
export class DeletedTask {
  taskId: string;
  taskName: string;
  boardId: string;
  userId: string;
  taskDescription: string;
  taskDeadline: string;
  assigneeEmail: string;
  initialState: string;


  constructor(
    taskId: string,
    taskName: string,
    boardId: string,
    userId: string,
    taskDescription: string,
    taskDeadline: string,

    assigneeEmail: string,
    initialState: string
    
  ) {
    this.taskId = taskId;
    this.taskName = taskName;
    this.boardId = boardId;
    this.userId = userId;
    this.taskDescription = taskDescription;
    this.taskDeadline = taskDeadline;
    this.assigneeEmail = assigneeEmail;
    this.initialState = initialState;

  }
}

File: Kanban-Board/TaskService/src/main/java/com/niit/TaskService/domain/DeletedTask.java
package com.niit.TaskService.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "deleted_tasks") // Specify the collection name if needed
public class DeletedTask {
    @Id
    private String taskId;
    private String taskName;
    private String boardId;
    private String userId;
    private String taskDescription;
    private String taskDeadline;
    private String assigneeEmail;
    private String initialState; // Added field to indicate initial state
    private List<String> todo = new ArrayList<>();
    private List<String> inProgress = new ArrayList<>();
    private List<String> completed = new ArrayList<>();
    private List<String> archived = new ArrayList<>();
    private LocalDateTime deletionTime; // Timestamp indicating when the task was deleted
}